jwst/jump/jump_step.py

```diff
@@ -30,8 +30,8 @@ def process(self, input):
             tstart = time.time()
             # Check for an input model with NGROUPS<=2
             ngroups = input_model.data.shape[1]
-            if ngroups <= 2:
-                self.log.warning('Can not apply jump detection when NGROUPS<=2;')
+            if ngroups <= 4:
+                self.log.warning('Can not apply jump detection when NGROUPS<=4;')
```

Comment:
Can you walk me through the logic again that says we need more than 4 groups? I know you can't detect a jump with only 2 differences (3 groups), but isn't it enough to have 3 differences (4 groups)?

Reply:
The algorithm uses the median of the first differences after excluding the largest difference. To get a reliable median you need three samples. So, adding in the difference that is excluded, we need four differences for the algorithm to work. This means we need five frames.
Making a change so that four frames gives a good result would not be hard.  My goal with this change was to make simple change first and at least match the input minimum frames to the algorithm minimum frames. This change is low risk (except for the cases of three and four frame integrations where the expected results could change)

Reply:
Ah, right - I forgot about the exclusion of the largest diff, thus requiring one more.